### PR TITLE
new mode, custom

### PR DESCRIPTION
**Pull request by RajRohitYadav**
Custom mode provides user option to select week length

passing `weekStartsOn` and `weekEndsOn` with `custom` mode will give desired week length.

I have taken care of reverse case, when `weekEndsOn < weekStartsOn`

Screenshot: 
https://user-images.githubusercontent.com/12390240/114059979-8cdfa600-98b2-11eb-981f-124851845462.png

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -168,3 +168,47 @@ export function getStyleForOverlappingEvent(
...
+    return daysCount;
+  } 
+  // normal week
+  else if (weekEndsOn > weekStartsOn) {
+    return weekEndsOn - weekStartsOn + 1
+  } 
+  // default
+  else {
+    return 1;
+  }
```
Comment (acro5piano) [inline src/utils.ts:213]:
```suggestion
  // handle reverse week
  if (weekEndsOn < weekStartsOn) {
    let daysCount = 1;
    let i = weekStartsOn;
    while(i !== weekEndsOn){
      ++i;
      if(i > 6) {
        i = 0;
      }
      ++daysCount;
      // fallback for infinite
      if(daysCount > 7){
        break;
      }
    }
    return daysCount;
  } 

  // normal week
  if (weekEndsOn > weekStartsOn) {
    return weekEndsOn - weekStartsOn + 1
  }

  // default
  return 1;
```

File: src/utils.ts [hunk 2]
```diff
@@ -168,3 +168,47 @@ export function getStyleForOverlappingEvent(
...
+) {
+  const subject = dayjs(date)
+  const subjectDOW = subject.day()
+  const days = Array(weekDaysCount(weekStartsOn,weekEndsOn))
+    .fill(0)
+    .map((_, i) => {
+      return subject.add(i - subjectDOW + weekStartsOn, 'day').locale(locale)
+    })
+  return days
+}
```
Comment (acro5piano) [inline src/utils.ts:186]: Can you write a unit test for this? I think it's easier to write unit tests for these kind of code.

**Comment by acro5piano** [comment]
@RajRohitYadav
Thank you for your PR! I would like to merge it, so first can you fix the Linting error?
Also it would be great if you

- Resolve my review
- Please run `yarn format` (it is pre-commit hook, but for some reason it isn't working)

File: src/utils.ts (hunk 2 above)
Comment (RajRohitYadav) [reply src/utils.ts]: I have to look into it, i will do it asap.

**Comment by acro5piano** [comment]
@RajRohitYadav 
Thanks for the update! Can you also run `yarn tslint --fix` to fix the lint error?

File: src/utils.ts (hunk 2 above)
Comment (acro5piano) [reply src/utils.ts]: Thanks!

**Comment by acro5piano** [comment]
@RajRohitYadav 
Sorry for too many mentions! Can you also add a story to the Storybook? Just add a component to this file.

https://github.com/acro5piano/react-native-big-calendar/blob/master/stories/desktop.stories.tsx

Thanks!

**Comment by RajRohitYadav** [comment]
> Nice work!! Can you fix a little bit?

It's done.

**Comment by RajRohitYadav** [comment]
Hey, sorry for the delay. I have added test cases and stories.

**Comment by acro5piano** [comment]
@RajRohitYadav Thanks for your contribution!

**Comment by acro5piano** [comment]
Released on v0.9.1.

**Pull request merged**